manta-crypto/src/arkworks/glv/mod.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Arkworks Elliptic Curve Implementation
+
+use crate::arkworks::{
+    ec::{
+        models::{
+            short_weierstrass_jacobian, twisted_edwards_extended, SWModelParameters,
+            TEModelParameters,
+        },
+        AffineCurve, ProjectiveCurve,
+    },
+    ff::PrimeField,
+};
+use alloc::{vec, vec::Vec};
+use ark_ff::{BigInteger, Field};
+use core::str::FromStr;
+use num_bigint::{BigInt, BigUint, Sign};
+
+#[cfg(feature = "ark-bls12-381")]
+use crate::arkworks::bls12_381;
+
+#[cfg(feature = "ark-bn254")]
+use crate::arkworks::bn254;
+
+/// Affine Curve Extension
+pub trait AffineCurveExt: AffineCurve {
```

Comment:
Actually, I forgot to mention, this trait (and its implementations) should go in `manta-crypto/src/arkworks/ec.rs` (and that file should re-export `ark-ec` the same way that `manta-crypto/src/arkworks/ff.rs` re-exports `ark-ff`).